core/src/main/scala/tamer/Tamer.scala

```diff
@@ -52,8 +52,11 @@ object Tamer {
 
   case class Live[K, V, S](
       config: KafkaConfig,
-      setup: Setup[K, V, S],
-      stateTransitionFunction: (S, Queue[Chunk[(K, V)]]) => ZIO[Any, TamerError, S],
+      serdes: Setup.Serdes[K, V, S],
+      defaultState: S,
+      stateHash: Int,
+      iteration: (S, Queue[Chunk[(K, V)]]) => Task[S],
```

Comment:
note how `iteration` has now lost the `R` (see [here](https://github.com/laserdisc-io/tamer/pull/390/files#diff-7567bcb2186e1226440531dc538c21790e90e5ff52e210ba92e00e79ebc33bbeR250-R251)) and is no longer erroring out with a `TamerError` necessarily. Actually, being in user space, it's likely we never should expect that to be the case